ebpf_extensions/neteventebpfext/netevent_ebpf_ext_event.c

```diff
@@ -74,15 +74,16 @@ typedef struct netevent_ext_header
 typedef struct netevent_ext_function_addresses
 {
     netevent_ext_header_t header;
+    netevent_capture_type_t capture_type;
     uint32_t helper_function_count;
     uint64_t* helper_function_address;
 } netevent_ext_function_addresses_t;
 
 // Dispatch table for the client module's helper functions
 static const void* _ebpf_netevent_ext_helper_functions[] = {(void*)&_ebpf_netevent_push_event};
-const netevent_ext_function_addresses_t _netevent_client_dispatch = {
-    .header =
-        {.version = EBPF_HELPER_FUNCTION_ADDRESSES_CURRENT_VERSION, .size = sizeof(netevent_ext_function_addresses_t)},
+netevent_ext_function_addresses_t _netevent_client_dispatch = {
+    .header = {.version = 2, .size = sizeof(netevent_ext_function_addresses_t)},
```

Comment:
When the client is using version 2, it needs to make sure the provider supports it. There is a bug in the old provider where it will attach to any client, regardless of version and if it can support the client version. To workaround this issue, clients using version 2 should check in their provider attach routine that the provider supplies a non-null NpiSpecificCharacteristics. See the change to PmnecsClientAttachProvider in my pktmon edge traversal PR